internal/cluster/common.go

```diff
@@ -259,37 +259,29 @@ func MapHostsByStatus(c *common.Cluster) map[string][]*models.Host {
 	return mapHostsByStatus(c, "")
 }
 
-func UpdateMachineCidr(db *gorm.DB, cluster *common.Cluster, machineCidr string) error {
-	// In case of dual-stack clusters the autocalculation feature is not supported. That means
-	// as soon as we detect that current Machine Network configuration indicates we have such
-	// a cluster, the function stops its execution.
-	reqDualStack := network.CheckIfClusterIsDualStack(cluster)
-	if reqDualStack {
-		return nil
-	}
-
+func UpdateMachineCidr(db *gorm.DB, cluster *common.Cluster, machineCidr []string) error {
```

Comment:
This function is performing many DB operations that should be a single transaction, is the `db` instance received here a transaction?